src/app/admin/deleteUser/DeleteUserDialog.tsx

```diff
@@ -0,0 +1,81 @@
+import Button from "@mui/material/Button";
+import DeleteIcon from "@mui/icons-material/Delete";
+import React from "react";
+import { UserRow } from "@/app/admin/page";
+import styled from "styled-components";
+import Modal from "@/components/Modal/Modal";
+import { deleteUser } from "@/app/admin/adminActions";
+import OptionButtonsDiv from "@/app/admin/common/OptionButtonsDiv";
+import { SetAlertOptions } from "@/app/admin/common/SuccessFailureAlert";
+
+const DangerDialog = styled(Modal)`
+    & #deleteUserDialog {
+        background-color: ${(props) => props.theme.error};
+    }
+`;
+
+interface Props {
+    userToDelete: UserRow | null;
+    setUserToDelete: (user: UserRow | null) => void;
+    setAlertOptions: SetAlertOptions;
+}
+
+const DeleteUserDialog: React.FC<Props> = (props) => {
+    if (props.userToDelete === null) {
+        return <></>;
+    }
+
+    const onDeleteConfirm = async (): Promise<void> => {
+        const { error } = await deleteUser(props.userToDelete!.id);
+
+        if (!error) {
+            props.setAlertOptions({
+                success: true,
+                message: (
+                    <>
+                        User <b>{props.userToDelete!.email}</b> deleted successfully.
+                    </>
+                ),
+            });
+        } else {
+            props.setAlertOptions({
+                success: false,
+                message: <>Delete User Operation Failed</>,
+            });
+        }
+
+        props.setUserToDelete(null);
+    };
+
+    const onDeleteCancel = (): void => {
+        props.setUserToDelete(null);
+    };
+
+    return (
+        <DangerDialog
+            header="DELETE USER"
```

Comment:
please 'normal case' the headers and set CSS to uppercase it (for screen readers) 😄